lib_eio/unix/eio_unix.mli

```diff
@@ -84,6 +84,20 @@ module Stdenv : sig
       You can use the functions in {!Eio.Stdenv} to access these fields if you prefer. *)
 end
 
+module Sockopt : sig
+  type 'a t = 'a Eio_unix__Sockopt.t =
+   | SO_KEEPALIVE : bool t
+   | SO_REUSEADDR : bool t
+   | SO_REUSEPORT : bool t
+   | TCP_CORK : int t
+   | TCP_KEEPCNT : int t
+   | TCP_KEEPIDLE : int t
+   | TCP_KEEPINTVL : int t
```

Comment:
could you also include TCP_NODELAY?

Reply:
Yeah, I'll add the complete set and tests once we decide on the interface.

Reply:
On the interface: seems easy to use and I like the GADT approach for the options themselves.

I really like having just a single function to set socket options.

Reply:
added TCP_NODELAY